### PR TITLE
Gsoc weeks 3 and 4

### PR DESCRIPTION
**Pull request by WassCodeur**
- Add my week3 blog post
- Add my week3 blog post

### CODE REVIEW

File: docs/source/posts/2024/2024-06-26-week3-wachiou-bouraima.rst [hunk 1]
```diff
@@ -0,0 +1,74 @@
+WEEK 3: Refinements and Further Enhancements
```
Comment (robinroy03) [inline docs/source/posts/2024/2024-06-26-week3-wachiou-bouraima.rst:1]: image

`Title underline too short` for `Hello everyone,`. Can be fixed by adding an extra `-`.

File: docs/source/posts/2024/2024-06-26-week3-wachiou-bouraima.rst [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+- You can find more details and the implementation in my pull request: `https://github.com/fury-gl/fury/pull/888 <https://github.com/fury-gl/fury/pull/888>`_.
+
+
+What Happens Next?
+------------------
+
+For week 4, I plan to:
+
+* Continue refining the ``warn_on_args_to_kwargs`` decorator based on feedback of my Peer `Iñigo Tellaetxe Elorriaga <https://github.com/itellaetxe>`_, `Robin Roy <https://github.com/robinroy03>`_, `Kaustav Deka <https://github.com/deka27>`_, my guide: `Serge Koudoro <https://github.com/skoudoro>`_ and the others community members.
```
Comment (robinroy03) [inline docs/source/posts/2024/2024-06-26-week3-wachiou-bouraima.rst:64]: ".. feedback _from_ my _peers_ .."

".. and the _other_ community members .."

**Review by robinroy03 — COMMENTED** [review]
Good work Wachiou!

Please see the below comments.

File: docs/source/posts/2024/2024-06-26-week3-wachiou-bouraima.rst (hunk 2 above)
Comment (WassCodeur) [reply docs/source/posts/2024/2024-06-26-week3-wachiou-bouraima.rst:64]: Hi @robinroy03!

Thank you for your feedback

File: docs/source/posts/2024/2024-06-26-week3-wachiou-bouraima.rst [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+Welcome to the fourth week of my Google Summer of Code (GSoC) 2024 journey!
+This week I've been delving into the technical aspects of my project,
+focusing on the consistent application of the ``warn_on_args_to_kwargs`` decorator and the initial implementation of lazy loading.
+
+
+Consistent application of ``warn_on_args_to_kwargs``
+----------------------------------------------------
+
+This week I continued to apply the decorator to foonctions.
+To ensure consistency across the code base,I audited all functions that could benefit from the ``warn_on_args_to_kwargs`` decorator.
```
Comment (itellaetxe) [inline docs/source/posts/2024/2024-06-26-week3-wachiou-bouraima.rst:21]: Need whitespace after comma:
`...the code base,I audited...` --> `...the code base, I audited...`

File: docs/source/posts/2024/2024-06-26-week3-wachiou-bouraima.rst [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+- You can find more details and the implementation in my pull request: `https://github.com/fury-gl/fury/pull/888 <https://github.com/fury-gl/fury/pull/888>`_.
+
+
+What Happens Next?
+------------------
+
+For week 4, I plan to:
+
+* Continue refining the ``warn_on_args_to_kwargs`` decorator based on feedback from my Peers `Iñigo Tellaetxe Elorriaga <https://github.com/itellaetxe>`_, `Robin Roy <https://github.com/robinroy03>`_, `Kaustav Deka <https://github.com/deka27>`_, my guide: `Serge Koudoro <https://github.com/skoudoro>`_ and the other community members.
+* Apply the ``warn_on_args_to_kwargs`` decorator to all the remaining modules and update all the unitests of these modules too, to respect the desired format.
```
Comment (itellaetxe) [inline docs/source/posts/2024/2024-06-26-week3-wachiou-bouraima.rst:65]: `unitests` --> `unit tests`

File: docs/source/posts/2024/2024-06-26-week3-wachiou-bouraima.rst [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+* Dive deep intp the lazy loading functionality based on my research to optimize performance.
```
Comment (itellaetxe) [inline docs/source/posts/2024/2024-06-26-week3-wachiou-bouraima.rst:66]: `intp` --> `into`

File: docs/source/posts/2024/2024-06-26-week3-wachiou-bouraima.rst [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+* Further engage in code reviews to support my peers and improve our project
```
Comment (itellaetxe) [inline docs/source/posts/2024/2024-06-26-week3-wachiou-bouraima.rst:67]: end this sentence with a period

File: docs/source/posts/2024/2024-06-26-week3-wachiou-bouraima.rst [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+Did I get stuck?
+----------------
+
+I didn't get stuck
```
Comment (itellaetxe) [inline docs/source/posts/2024/2024-06-26-week3-wachiou-bouraima.rst:72]: end this sentence with a period

File: docs/source/posts/2024/2024-06-26-week4-wachiou-bouraima.rst [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+Updating the ``warn_on_args_to_kwargs`` decorator
+-------------------------------------------------
+
+Based on feedback from my mentor `Serge Koudoro <https://github.com/skoudoro>`_  and peers  `Iñigo Tellaetxe Elorriaga <https://github.com/itellaetxe>`_, `Robin Roy <https://github.com/robinroy03>`_, `Kaustav Deka <https://github.com/deka27>`_, I refined the ``warn_on_args_to_kwargs`` decorator and its associated unit tests:
+
+1. Improvements:
+
+   - Added conditions to verify if the values of ``from_version``, ``until_version``, and the current version of FURY are respected. This includes handling cases where ``from_version`` is greater than the current version of FURY, ``until_version`` is less than the current version of FURY, and ``until_version`` is greater than or equal to the current version of FURY.
+   - Ensured the decorator and tests cover a broader range of edge cases.
+   - Enhanced the warnong messages for better clarity and guidance.
```
Comment (itellaetxe) [inline docs/source/posts/2024/2024-06-26-week4-wachiou-bouraima.rst:25]: `warnong` --> `warning`

File: docs/source/posts/2024/2024-06-26-week4-wachiou-bouraima.rst [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+
+
+Exploring lazy loading
+----------------------
+
+In order to optimize performance, I've started exploring and implementing lazy loading. This week, the focus was on the following points:
+
+- Getting to grips with how the lazy loader works
+- Implementing some small script to understand how the Lazy loader works
+- I also read the SPEC1 document available at `SPEC1 <https://scientific-python.org/specs/spec-0001/>`_.
```
Comment (itellaetxe) [inline docs/source/posts/2024/2024-06-26-week4-wachiou-bouraima.rst:77]: Remove the period on this sentence, or end the rest of the points with a period

File: docs/source/posts/2024/2024-06-26-week4-wachiou-bouraima.rst [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+- `ui`: https://github.com/fury-gl/fury/pull/902
+
+
+Exploring lazy loading
+----------------------
+
+In order to optimize performance, I've started exploring and implementing lazy loading. This week, the focus was on the following points:
+
+- Getting to grips with how the lazy loader works
+- Implementing some small script to understand how the Lazy loader works
```
Comment (itellaetxe) [inline docs/source/posts/2024/2024-06-26-week4-wachiou-bouraima.rst:76]: `Lazy` --> `lazy`

**Review by itellaetxe — APPROVED** [review]
LGTM Wachiou. I only found spelling issues.
On the other hand, I am curious to see how you implement lazy loading. Good work!

**Comment by m-agour** [comment]
Hi @WassCodeur , please address @itellaetxe review above.

File: docs/source/posts/2024/2024-06-26-week4-wachiou-bouraima.rst (hunk 10 above)
Comment (WassCodeur) [reply docs/source/posts/2024/2024-06-26-week4-wachiou-bouraima.rst]: Hello @itellaetxe , 

thank you very much for your very informative comments. I'm sorry for the delay in responding.

**Review by skoudoro — COMMENTED** [review]
All good. merging
